terraform/modules/redis/main.tf

```diff
@@ -0,0 +1,54 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.0"
+    }
+  }
+}
+
+/// elasticache cluster
+
+resource "aws_elasticache_subnet_group" "redis_cluster_subnet_group" {
+  name       = var.cluster_name
+  subnet_ids = var.subnet_ids
+}
+
+resource "aws_security_group" "redis" {
+  name        = "${var.cluster_name}_elasticache"
```

Comment:
Do you have a plan for how this will work with multiple test environments?

Reply:
are you referring to the workspaces work? This is not merge yet.

Reply:
It doesn't matter whether that work is merged yet. The reason I'm asking is because your colleagues are working on enabling multiple test environments per account, so it would be considerate to think about whether your PR is compatible with that.

I think all you need to do to ensure that this plays nicely with multiple copies of the infrastructure is to remove the default from `cluster_name` in the Redis module and pass it in from the govuk module. Someone can then easily add the environment name to it later once there's a variable for that.

Reply:
I will discuss with Roch, I guess I need to pull the he cluster name variable up to deployment level.